src/Symfony/Bridge/Twig/Resources/views/Form/tailwind_2_layout.html.twig

```diff
@@ -0,0 +1,82 @@
+{% extends 'form_div_layout.html.twig' %}
+
+{%- block widget_container_attributes -%}
+    {%- set attr = attr|merge({ class: attr.class|default(block('row_class')) }) -%}
```

Comment:
using a block to set the value of the tag has a huge drawback: double escaping. Any variable you echo in the block will be escaped when echoing it. And the class attribute will be escaped when rendering it.

Reply:
Can you show me an example of what you mean? I tried:

```twig
{% block widget_class %}{{ parent() }} {{ 'bg-green-300' }}{% endblock %}
```

and it works as expected.

Reply:
you are not using a variable

Reply:
Sorry, I'm still not seeing the issue. I tried this as well:

```twig
{% block row_class %}{% set var = 'md:bg-green-500' %}{{ parent() }} {{ var }}{% endblock %}
```

Reply:
well, it could cause surprising issues if you have cases where a class name includes some characters that get HTML escaped. that's not the case in tailwindcss, but I already saw other libraries doing that (I don't remember the name right now), and using such pattern here might make people think that it is fine doing the same in their code, without knowing the limitation.

Reply:
Ok, understood and yes, I suppose a user could add their own non-tailwind-specific classes here as well.

Do we not have the same problem in `form_div_layout.html.twig`? (ie `block('widget_attribute')` which calls `block('attributes')`). This is where I got the idea.

Reply:
no, because there, the result of calling `block()` is displayed directly, and `block()` is marked as a safe function regarding autoescaping. But in this code, you don't display the block. You assign it inside a property of the array.

Reply:
Ah, ok. What about instead of the "class" blocks, having block specific class variables that can be overridden by a user's custom theme:

```twig
{# core layout #}
{%- block widget_container_attributes -%}
    {%- set attr = attr|merge({ class: attr.class|default(row_class|default('space-y-6')) }) -%}
    {{- parent() -}}
{%- endblock widget_container_attributes -%}

{# user's extended layout #}
{%- block widget_container_attributes -%}
    {%- set row_class = 'space-y-6 bg-green-300' -%}
    {{- parent() -}}
{%- endblock widget_container_attributes -%}
```

The `row_class` var will be escaped, but not double escaped.

Reply:
The "class variable" way has the added benefit of being able to be overridden with the form twig functions:

```twig
{{ form_widget(form, {widget_class: 'mt-1 w-full bg-green-200'}) }}
```

Reply:
I'm going to remove the class blocks and just use variables.